### PR TITLE
Default to edit for buffers visiting files

### PR DESCRIPTION
**Pull request by deb0ch**
As discussed in #113, implementing tests for this is still needed.

I tested it manually and

- the buffers that usually open in `'edit` are still in `'edit`
- `*magit: blah*` still gets `'general`
- if I create a file (`SPC ff RET blablabla RET` in Spacemacs it nicely gets `'edit`
- if I create a buffer (`spacemacs/new-empty-buffer`) with a random name it nicely gets `'general`

### CODE REVIEW

**Comment by deb0ch** [comment]
Okay, actually I have figured out how to write the tests, but unfortunately I couldn't test the tests.

I had an error:

```
cask exec: error: Failed to execute ert-runner: [Errno 2] No such file or directory
```

I guess it is a macos compatibility issue. I have to go now, I don't have time anymore to fix this, if you could test the tests and tell me it would be great ! 👍✨

**Comment by deb0ch** [comment]
Hahaha, what made the test fail previously was... the other test, not the commit itself.

It's all clean now, ready for proper review and merge (?).

Indeed, implementing the tests was a *lot* harder than implementing the feature 💦

File: window-purpose.el [hunk 1]
```diff
@@ -165,13 +165,16 @@ This allows Purpose to work well with both `ido' and `helm'.")
 
 ;;; Overloaded commands: (C-u to get original Purpose-less behavior)
 (define-purpose-prefix-overload purpose-find-file-overload
-  '(purpose-friendly-find-file find-file-without-purpose))
+  '(purpose-friendly-find-file
+    find-file-without-purpose))
```
Comment (bmag) [inline window-purpose.el]: Unnecessary formatting changes.

File: window-purpose.el [hunk 2]
```diff
@@ -165,13 +165,16 @@ This allows Purpose to work well with both `ido' and `helm'.")
...
 
 (define-purpose-prefix-overload purpose-find-file-other-window-overload
-  '(purpose-friendly-find-file-other-window find-file-other-window-without-purpose))
+  '(purpose-friendly-find-file-other-window
+    find-file-other-window-without-purpose))
```
Comment (bmag) [inline window-purpose.el]: Unnecessary formatting changes.

File: window-purpose.el [hunk 3]
```diff
@@ -165,13 +165,16 @@ This allows Purpose to work well with both `ido' and `helm'.")
...
 
 (define-purpose-prefix-overload purpose-find-file-other-frame-overload
-  '(purpose-friendly-find-file-other-frame find-file-other-frame-without-purpose))
+  '(purpose-friendly-find-file-other-frame
+    find-file-other-frame-without-purpose))
```
Comment (bmag) [inline window-purpose.el]: Unnecessary formatting changes.

File: window-purpose-core.el [hunk 4]
```diff
@@ -212,6 +212,9 @@ If no purpose was determined, return `default-purpose'."
      (purpose--buffer-purpose-mode buffer-or-name
                                    purpose--default-mode-purposes)))
 
+   ;; If the buffer is visiting a file, fallback to 'edit purpose
+   (when (buffer-file-name (get-buffer buffer-or-name))
+     'edit)
```
Comment (bmag) [inline window-purpose-core.el]: Should use a new custom `default-file-purpose` variable, which has a default value of `edit`.

File: test/core-test.el [hunk 5]
```diff
@@ -261,6 +261,23 @@ The prompt is chosen according to `purpose-preferred-prompt'."
...
+   (find-file "foo")
+   (should (equal (purpose-buffer-purpose (get-buffer "foo")) 'bar))
+   (kill-buffer "foo")
+   (find-file "baz")
+   (should (equal (purpose-buffer-purpose (get-buffer "baz")) 'edit))
+   (kill-buffer "baz")
+   (get-buffer-create "yolo")
+   (should (equal (purpose-buffer-purpose (get-buffer "yolo")) default-purpose))
+   (kill-buffer "yolo")))
+
```
Comment (bmag) [inline test/core-test.el]: Use `unwind-protect` and `purpose-kill-buffers-safely`. This ensures proper tear-down of the test, and separate the tear-down from what's actually being tested. See other tests in the same file for examples.

**Comment by deb0ch** [comment]
All done 👌✨

**Comment by bmag** [comment]
I squashed your commits and made some additional fixes in https://github.com/bmag/emacs-purpose/commit/2655bbe3399f00d3297ded58f92e7be22876148a. Thank you.

**Comment by deb0ch** [comment]
Yay !  🎉 🎊 

That was a swift one, so good to see something merged so quickly right after initial discussion 😸

Nice work !
